statistician.py

```diff
@@ -1,209 +1,114 @@
-                           3: self.won_with_methods_stats, 4: self.won_with_methods_per_round_stats,
-                           5: self.lost_with_methods_stats, 6: self.lost_with_methods_per_round_stats,
-                           7: self.points_earned_with_method_stats}
-
-        self.__bot1_name = bot1_name
-        self.__bot2_name = bot2_name
-        self.__game_id = game_id
-        self.__round = 0
-        self.__bank = {GENERAL: StatisticBank(GENERAL, self),
-                       self.__bot1_name: StatisticBank(self.__bot1_name, self),
-                       self.__bot2_name: StatisticBank(self.__bot2_name, self)}
-
-        self.__bank[GENERAL].round_won_by[DRAW] = 0
-        self.__bank[GENERAL].round_won_by[self.__bot1_name] = 0
-        self.__bank[GENERAL].round_won_by[self.__bot2_name] = 0
-
-        self.__bank[GENERAL].advantage_after_round[TIME] = 0
-        self.__bank[GENERAL].advantage_after_round[self.__bot1_name] = 0
-        self.__bank[GENERAL].advantage_after_round[self.__bot2_name] = 0
-
-        self.__stats = {'round_won_by': self.__bank[GENERAL].round_won_by,
-                        'advantage_after_round': self.__bank[GENERAL].advantage_after_round,
-                        'method_used': self.method_used_stats,
-                        'method_used_per_round': self.methods_used_per_round_stats,
-                        'won_with_methods': self.won_with_methods_stats,
-                        'won_with_methods_per_round': self.won_with_methods_per_round_stats,
-                        'lost_with_methods': self.lost_with_methods_stats,
-                        'lost_with_methods_per_round': self.lost_with_methods_per_round_stats,
-                        'points_earned_with_method': self.points_earned_with_method_stats}
-
-    def __updateRoundWonBy(self, winner):
-        """
-        Private helper method used to updating round winner statistics
-        :param winner: RoundWinner from enumeration package with information about round winner
-        """
-        if winner is rw.DRAW:
-            self.__bank[GENERAL].round_won_by[DRAW] += 1
-        elif winner is rw.BOT_1:
-            self.__bank[GENERAL].round_won_by[self.__bot1_name] += 1
-        else:
-            self.__bank[GENERAL].round_won_by[self.__bot2_name] += 1
 
-    def __updateWonLostByMethod(self, bot1_method, bot2_method, winner):
+    @staticmethod
+    def __generateLeaderboard():
+        styles = """
+        <style>
+        table {
+          font-family: arial, sans-serif;
+          border-collapse: collapse;
+          width: 100%;
+        }
+
+        th {
+          border: 1px solid #000000;
+          text-align: center;
+          color: #616af2;
+          background-color: #000000;
+          padding: 8px;
+        }
+
+        td {
+          border: 1px solid #000000;
+          text-align: left;
+          padding: 8px;
+        }
```

Comment:
I think that we can move this whole style block to same other global class/file.
Just like you store css in other file than html.
So that they do not obscure the html generation process

Reply:
I don't completely agree with creating another file just for styles. 
However, sometimes it is better to have peace of mind than to have right so I am changing it :D